src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -0,0 +1,695 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace System.Collections.Generic
+{
+    /// <summary>
+    /// Represents a data structure in which each element additionally has a priority
+    /// associated with it.
+    /// </summary>
+    /// <typeparam name="TElement">The type of the element.</typeparam>
+    /// <typeparam name="TPriority">The type of the priority.</typeparam>
+    public class PriorityQueue<TElement, TPriority>
+    {
+        /// <summary>
+        /// Represents an implicit heap-ordered complete d-ary tree, stored as an array.
+        /// </summary>
+        private (TElement element, TPriority priority)[] _nodes;
+
+        private Lazy<UnorderedItemsCollection> _unorderedItems;
```

Comment:
This should just be:
```C#
private UnorderedItemsCollection? _unorderedItems;
```

The line:
```C#
_unorderedItems = new Lazy<UnorderedItemsCollection>(() => new UnorderedItemsCollection(this));
```
should be deleted from the ctors.

And the property:
```C#
public UnorderedItemsCollection UnorderedItems => _unorderedItems.Value;
```
should be changed to be:
```C#
public UnorderedItemsCollection UnorderedItems => _unorderedItems ??= new UnorderedItemsCollection(this);
```

Reply:
nice one :D